research/object_detection/exporter_lib_v2.py

```diff
@@ -75,11 +99,34 @@ def _run_inference_on_images(self, image):
 class DetectionFromImageModule(DetectionInferenceModule):
   """Detection Inference Module for image inputs."""
 
-  @tf.function(
-      input_signature=[
-          tf.TensorSpec(shape=[1, None, None, 3], dtype=tf.uint8)])
-  def __call__(self, input_tensor):
-    return self._run_inference_on_images(input_tensor)
+  def __init__(self, detection_model,
+               use_side_inputs=False,
+               zipped_side_inputs=None):
+    """Initializes a module for detection.
+
+    Args:
+      detection_model: the detection model to use for inference.
+      use_side_inputs: whether to use side inputs.
+      zipped_side_inputs: the zipped side inputs.
+    """
+    self.side_input_names = []
+    sig = [tf.TensorSpec(shape=[1, None, None, 3], dtype=tf.uint8)]
+    if use_side_inputs:
+      for info in zipped_side_inputs:
+        self.side_input_names.append(info[2])
+        sig.append(tf.TensorSpec(shape=info[0],
+                                 dtype=info[1],
+                                 name=info[2]))
+
+    def call_func(self, input_tensor, *side_inputs):
+      kwargs = dict(zip(self.side_input_names, side_inputs))
+      return self._run_inference_on_images(input_tensor, **kwargs)
+
+    self.__call__ = tf.function(self.call_func, input_signature=sig)
```

Comment:
According to this
https://www.tensorflow.org/api_docs/python/tf/saved_model/save

function signatures are optional. Can you run a quick test to see what happens when you omit them?

Reply:
Another thought. If the user specifies tensor names from the corresponding dataset, can we infer shapes and types from there ?

Reply:
In order to create the saved model we need tensorspecs, doesn't work without it (at least within the limits of how we're diong it now). How would we infer shapes? Tensorspecs require shape as an argument so don't we need them at model-save-time? Also, the Tf1 exporter requires all three of these arguments so I sort of modeled after that. Would appreciate your thoughts.

Reply:
1. We have access to the dataset, can we use that to infer the shape and types ? 
2. According to the documentation I linked above, function signatures are optional. If removing them (omitting the input signatures argument) gives an error, it world be good to have a record of it and clarify it with the tensorflow team for future reference.

Reply:
Talked with @tombstone, will leave this as a to-do.